src/test/unit/services/pathfinder/util.hpp

```diff
@@ -0,0 +1,472 @@
+#ifndef TEST_UNIT_SERVICES_PATHFINDER_UTIL_HPP
+#define TEST_UNIT_SERVICES_PATHFINDER_UTIL_HPP
+
+#include <stan/callbacks/interrupt.hpp>
+#include <stan/callbacks/stream_writer.hpp>
+#include <iostream>
+
+namespace stan {
+namespace test {
+
+struct mock_callback : public stan::callbacks::interrupt {
+  int n;
+  mock_callback() : n(0) {}
+
+  void operator()() { n++; }
+};
+
+class loggy : public stan::callbacks::logger {
```

Comment:
I would have named it something other than 'loggy' and put it somewhere other than in the pathfinder utils, but I think it's OK like this for now.  I created an issue for this one and the next:  https://github.com/stan-dev/stan/issues/3194